src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -129,7 +143,21 @@ function WorkspaceCategoriesPage({policyCategories, route}: WorkspaceCategoriesP
                     <View style={[styles.ph5, styles.pb5]}>
                         <Text style={[styles.textNormal, styles.colorMuted]}>{translate('workspace.categories.subtitle')}</Text>
                     </View>
-                    {categoryList.length ? (
+                    {!isOffline && policyCategories === undefined && (
+                        <ActivityIndicator
+                            size={CONST.ACTIVITY_INDICATOR_SIZE.LARGE}
+                            style={[styles.flex1]}
+                            color={theme.textSupporting}
+                        />
+                    )}
+                    {(isOffline || Boolean(policyCategories)) && categoryList.length === 0 && (
+                        <WorkspaceEmptyStateSection
+                            title={translate('workspace.categories.emptyCategories.title')}
+                            icon={Illustrations.EmptyStateExpenses}
+                            subtitle={translate('workspace.categories.emptyCategories.subtitle')}
+                        />
+                    )}
+                    {(isOffline || Boolean(policyCategories)) && Boolean(categoryList.length) && (
```

Comment:
Define `const isLoading = !isOffline && policyCategories === undefined`
And change rendering conditions to:

1. categoryList.length === 0 && isLoading
2. categoryList.length === 0 && !isLoading
3. categoryList.length > 0